package/MDAnalysis/auxiliary/XVG.py

```diff
@@ -0,0 +1,245 @@
+        # '#' can be anywhere in the line, everything after is a comment
+        comment_position = line.find('#')
+        if comment_position > 0 and line[:comment_position]:
+            yield line[:comment_position]
+        elif comment_position < 0 and line:
+            yield line
+        # if comment_position == 0, then the line is empty
+
+
+class XVGReader(base.AuxReader):
+    """ Auxiliary reader to read data from an .xvg file.
+
+    All data from the file will be read and stored on initialisation. This is
+    the default reader for .xvg files.
+    
+    xvg files are produced by Gromacs during simulation or analysis, formatted
+    for plotting data with Grace.
+
+    Parameters
+    ----------
+    filename : str
+       Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.    
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+
+    Note
+    ----
+    The time of each step is assumed to stored in the first column (``time_col``
+    defaults to 0), in units of ps, and data is assumed to be time-ordered.
+
+    The file is assumed to be of a size such that reading and storing the full 
+    contents is practical.
+    """
+
+    format = "XVG"
+
+    def __init__(self, filename, **kwargs):
+        self._data_input = os.path.abspath(filename)
+        with open(filename) as xvg_file:
+            lines = xvg_file.readlines()
+        auxdata = []
+        # remove comments before storing
+        for i, line in enumerate(uncomment(lines)):
+           auxdata.append([float(l) for l in line.split()])
+           # check the number of columns is consistent
+           if len(auxdata[i]) != len(auxdata[0]):
+                raise ValueError('Step {0} has {1} columns instead of '
+                                 '{2}'.format(i, auxdata[i], auxdata[0]))
+        self._auxdata = np.array(auxdata)
+        # default to time in first column, if not otherwise specified
+        time_col = kwargs.pop('time_col', 0)
+        super(XVGReader, self).__init__(time_col=time_col, **kwargs)
+
+    def _read_next_step(self):
+        """ Read next auxiliary step. """
+        new_step = self.step + 1
+        if new_step < self.n_steps:
+            self._data = self._auxdata[new_step]
+            self.step = new_step
+        else:
+            self.go_to_first_step()
+            raise StopIteration
+
+    def go_to_step(self, i):
+        """ Move to and read i-th auxiliary step. 
+
+        Parameters
+        ----------
+        i : int
+            Step number (0-indexed) to move to
+
+        Raises
+        ------
+        ValueError
+            If step index not in valid range.
+        """
+        if i not in range(self.n_steps):
+            raise ValueError("Step index {0} is not valid for auxiliary "
+                             "(num. steps {1})".format(i, self.n_steps))
+        self.step = i-1
+        self.next()
+
+    def count_n_steps(self):

```

Comment:
The reason I did this is so I could define the `n_steps` property in the base class, and each reader could either set it directly in `__init__` or define a `count_n_steps` that's called to set `n_steps` if it isn't already. The latter wasn't really needed for the default XVG reader - I can just set directly in `__init__`. Also, since I've defined `__len__` in the base class, it doesn't really serve a purpose to the user; I'll probably switch it to a `_count_n_steps`.

Ultimately, I think I mainly did this to make `__init__` cleaner, and mimicking what some of the trajectory readers were doing,  but it may not be necessary at all. 

I started calling them 'steps' rather than 'frames' to better distinguish the two, and made it seem a bit nicer to me; I can see what you mean, though. It's now propagated through quite a lot of the code, but I could still think about changing it?